bokeh/core/has_props.py

```diff
@@ -529,21 +529,23 @@ def query_properties_with_values(self, query, include_defaults=True):
             dict : mapping of property names and values for matching properties
 
         '''
+        themed_keys = set()
         result = dict()
         if include_defaults:
             keys = self.properties()
         else:
             keys = set(self._property_values.keys())
             if self.themed_values():
-                keys |= set(self.themed_values().keys())
+                themed_keys = set(self.themed_values().keys())
+                keys |= themed_keys
 
         for key in keys:
-            prop = self.lookup(key)
-            if not query(prop):
+            descriptor = self.lookup(key)
+            if not query(descriptor):
                 continue
 
-            value = prop.serializable_value(self)
-            if not include_defaults:
+            value = descriptor.serializable_value(self)
+            if not include_defaults and key not in themed_keys:
```

Comment:
Unclear to me whether themed values should be considered default or not. On the one hand, theming is explicitly for setting "new defaults" but BokehJS knows nothing about them unless they are sent. In that sense I think this this needs to always report themed values, independent of what `include_defaults` is set to.